### ISSUE TITLE
WSS Server: java.util.concurrent.RejectedExecutionException on connection

### ISSUE THREAD

**Issue by diabolusss** [TooTallNate/Java-WebSocket#270]
I'm trying to run a wss server on localhost using unchanged SSLServerExample from trunk. After connecting to it from [websocket.org](http://www.websocket.org/echo.html) or after a while it crashes with this exception:

```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@1ea9246 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@1293ce7[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 4]
    at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2048)
    at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:821)
    at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:325)
    at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:530)
    at java.util.concurrent.ScheduledThreadPoolExecutor.submit(ScheduledThreadPoolExecutor.java:629)
    at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:638)
    at org.java_websocket.SSLSocketChannel2.consumeDelegatedTasks(SSLSocketChannel2.java:176)
    at org.java_websocket.SSLSocketChannel2.processHandshake(SSLSocketChannel2.java:141)
    at org.java_websocket.SSLSocketChannel2.<init>(SSLSocketChannel2.java:83)
    at org.java_websocket.server.DefaultSSLWebSocketServerFactory.wrapChannel(DefaultSSLWebSocketServerFactory.java:39)
    at org.java_websocket.server.WebSocketServer.run(WebSocketServer.java:320)
    at java.lang.Thread.run(Thread.java:745)
```

I get the same error after client\server closes connection.
After that i can connect to socket server only after restarting it. 

It looks like a handshake processor crashes, because when i try to connect from websocket.org it says neither connected or dissconnected, but when i restart socket server in log appear messages "disconnected". like connections where awaiting handshake.

I would greatly appreciate any help you can give me.

**Comment by BrushfireDigitalServices** [comment]
This fix will be part of the next release, and has been added to the milestone. We're fixing bugs on master and confirming stability ahead of the 1.4.0 release.

Related: #315
Should be fixed by: #216 (partial) or #331 (complete)

Now is a good time to confirm that #331 works and that the workaround provided by @diabolusss is no longer needed. I will close this after a fix is merged.

**Comment by chico1198** [comment]
I implemented this fix, but I can still get this error in a certain situation.

My implementation wants only a single connection to the server. So, in my server code I have the following:
```
	@Override
	public void onOpen(WebSocket conn, ClientHandshake handshake ) {
		if(connections().size() > 1) {
			System.out.println("Connection request after first connection. Refusing request.");
			// Use custom code to inform client of refusal
			conn.closeConnection(1111, "Websocket connection count exceeded.");
		} else {
			System.out.println(conn + " connected. " +
					"Type characters and press 'return' to send them to attached client.");
		}
	}
```
This works as expected for the second connection. That is, the connection is refused. However, when a third or subsequent client attempts to connect, I get the same server error as noted above.
```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@16667c73 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@783835ee[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 4]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
	at java.util.concurrent.ScheduledThreadPoolExecutor.submit(ScheduledThreadPoolExecutor.java:632)
	at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:678)
	at org.java_websocket.SSLSocketChannel2.consumeDelegatedTasks(SSLSocketChannel2.java:176)
	at org.java_websocket.SSLSocketChannel2.processHandshake(SSLSocketChannel2.java:141)
	at org.java_websocket.SSLSocketChannel2.write(SSLSocketChannel2.java:208)
	at org.java_websocket.SSLSocketChannel2.writeMore(SSLSocketChannel2.java:351)
	at org.java_websocket.SocketChannelIOHelper.batch(SocketChannelIOHelper.java:49)
	at org.java_websocket.server.WebSocketServer.run(WebSocketServer.java:344)
	at java.lang.Thread.run(Thread.java:745)
```
Again, I *do* have the fix described above in place.

Is there any way to fix this?

**Comment by bendem** [comment]
@marci4 this is fixed.

**Comment by marci4** [comment]
@bendem thx for the response. Closing it
